src/index.js

```diff
@@ -37,7 +37,12 @@ function buildValidations(referenced, dereferenced, receivedOptions) {
     const schemas = {};
 
     const basePaths = dereferenced.servers && dereferenced.servers.length
-        ? dereferenced.servers.map(({ url }) => new URL(url).pathname)
+        ? dereferenced.servers.map(({ url, variables = {} }) => {
+            Object.keys(variables).forEach((key) => {
+                url = url.replace(new RegExp(`{${key}}`), variables[key].default || '');
```

Comment:
default is required and should fail the validator in case it's missing, so i don't think we should handle its absence here